aws/resource_aws_xray_sampling_rule.go

```diff
@@ -0,0 +1,212 @@
+	params := &xray.GetSamplingRulesInput{}
+
+	for {
+		out, err := conn.GetSamplingRules(params)
+		log.Printf("[DEBUG] Retrieved Rules: %s", out.SamplingRuleRecords)
+		if err != nil {
+			d.SetId("")
+			return err
+		}
+		for _, samplingRuleRecord := range out.SamplingRuleRecords {
+			samplingRule := samplingRuleRecord.SamplingRule
+			if aws.StringValue(samplingRule.RuleName) == d.Id() {
+				d.Set("rule_name", samplingRule.RuleName)
+				d.Set("resource_arn", samplingRule.ResourceARN)
+				d.Set("priority", samplingRule.Priority)
+				d.Set("fixed_rate", samplingRule.FixedRate)
+				d.Set("reservoir_size", samplingRule.ReservoirSize)
+				d.Set("service_name", samplingRule.ServiceName)
+				d.Set("service_type", samplingRule.ServiceType)
+				d.Set("host", samplingRule.Host)
+				d.Set("http_method", samplingRule.HTTPMethod)
+				d.Set("url_path", samplingRule.URLPath)
+				d.Set("version", samplingRule.Version)
+				d.Set("attributes", aws.StringValueMap(samplingRule.Attributes))
+				d.Set("created_at", samplingRuleRecord.CreatedAt)
+				d.Set("modified_at", samplingRuleRecord.ModifiedAt)
+				d.Set("rule_arn", samplingRule.RuleARN)
+				break
+			}
+		}
+		if out.NextToken == nil {
+			break
+		}
+		params.NextToken = out.NextToken
+	}
+	return nil
+}
+
+func resourceAwsXraySamplingRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).xrayconn
+	samplingRuleUpdate := &xray.SamplingRuleUpdate{
+		RuleName:      aws.String(d.Id()),
+		Priority:      aws.Int64(int64(d.Get("priority").(int))),
+		FixedRate:     aws.Float64(d.Get("fixed_rate").(float64)),
+		ReservoirSize: aws.Int64(int64(d.Get("reservoir_size").(int))),
+		ServiceName:   aws.String(d.Get("service_name").(string)),
+		ServiceType:   aws.String(d.Get("service_type").(string)),
+		Host:          aws.String(d.Get("host").(string)),
+		HTTPMethod:    aws.String(d.Get("http_method").(string)),
+		URLPath:       aws.String(d.Get("url_path").(string)),
+	}
+
+	if d.HasChange("version") {
+		return fmt.Errorf("Version cannot be modified")
+	}
+
+	if d.HasChange("attributes") {
+		attributes := map[string]*string{}
+		if v, ok := d.GetOk("attributes"); ok {
+			if m, ok := v.(map[string]interface{}); ok {
+				attributes = stringMapToPointers(m)
+			}
+		}
+		samplingRuleUpdate.Attributes = attributes
+	}
+
+	params := &xray.UpdateSamplingRuleInput{
+		SamplingRuleUpdate: samplingRuleUpdate,
+	}
+
+	_, err := conn.UpdateSamplingRule(params)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsXraySamplingRuleRead(d, meta)
+}
+
+func resourceAwsXraySamplingRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).xrayconn
+
+	log.Printf("[INFO] Deleting XRay Sampling Rule: %s", d.Id())
+
+	params := &xray.DeleteSamplingRuleInput{
+		RuleName: aws.String(d.Id()),
+	}
+	_, err := conn.DeleteSamplingRule(params)
+	if err != nil {
+		return fmt.Errorf("Error deleting XRay Sampling Rule: %s", d.Id())
```

Comment:
This error message should include the error as well 😄 

```suggestion
		return fmt.Errorf("error deleting XRay Sampling Rule (%s): %s", d.Id(), err)
```